component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,279 @@
+                            attributeExpressionExecutors[2]).getValue()));
+                } else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.LONG) {
+                    timeInMilliSeconds = Long.parseLong(String.valueOf(((ConstantExpressionExecutor)
+                            attributeExpressionExecutors[2]).getValue()));
+                } else {
+                    throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter timeout " +
+                            "should be either int or long, but found " + attributeExpressionExecutors[2]
+                            .getReturnType());
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException(
+                    "Resource batch window should only have two parameters, " + "but found "
+                            + attributeExpressionExecutors.length + " input attributes");
+        }
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner) {
+        synchronized (this) {
+            long currentTime = siddhiAppContext.getTimestampGenerator().currentTime();
+            if (nextEmitTime == -1) {
+                nextEmitTime = currentTime + timeInMilliSeconds;
+                if (scheduler != null) {
+                    scheduler.notifyAt(nextEmitTime);
+                }
+            }
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                int groupEventMapSize = 0;
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    if (groupEventMap.containsKey(groupEventMapKey)) {
+                        groupEventMap.get(groupEventMapKey).streamEventList.add(clonedStreamEvent);
+                    } else {
+                        List<StreamEvent> list = new ArrayList<>();
+                        list.add(clonedStreamEvent);
+                        groupEventMap.put(groupEventMapKey, new ResourceStreamEventList(list,
+                                clonedStreamEvent.getTimestamp() + timeInMilliSeconds));
+                    }
+                    groupEventMapSize = groupEventMap.get(groupEventMapKey).streamEventList.size();
+                }
+                for (Map.Entry<Object, ResourceStreamEventList> entry : groupEventMap.entrySet()) {
+                    if (groupEventMapSize == windowLength || entry.getValue().expiryTimestamp <
```

Comment:
I think checking groupEventMapSize here is wrong as it may not be for the the entry set you are looking at. 

need to use entry.getValue().list.size() is it ?

Reply:
resource identifiers can come and go so we need to get the windowLength dynamically and check for 
groupEventMapSize >= windowLength

Reply:
Fixes added: 290829b93bcce354bfba036e25b4774e4d45b374

With the new fix, before doing the above comparison the windowLength dynamically by invoking the Resource identifier static map and get the list size.